include/lsst/afw/cameraGeom/DetectorCollection.h

```diff
@@ -34,45 +34,29 @@ namespace lsst {
 namespace afw {
 namespace cameraGeom {
 
-/**
- * An immutable collection of Detectors that can be accessed by name or ID
- */
-class DetectorCollection : public table::io::PersistableFacade<DetectorCollection>,
-                           public table::io::Persistable {
+template <typename T>
+class DetectorCollectionBase {
```

Comment:
I don't object to having "Detector" in the name, but I still think that `DetectorBase` and `DetectorCollectionBase` are very unhelpful names because they don't say what the classes are (except that they each have a subclass with a much more natural meaning). It's the same kind of confusing split as `geom::Point` vs. `geom::PointBase`.

Please consider giving these two classes more descriptive names that make it clear what they represent (besides "common elements of two other classes").

Reply:
`DetectorCollectionBase` is used to add a (template) collection-of-detectors interface to its subclasses.  I actually think that's pretty descriptive.

`DetectorBase` really is just "stuff that's common to `Detector` and its `Builders`.  IMO the best name for it would be something that says "nothing to see here; just pretend all of the public interfaces here were directly duplicated in subclasses".  Maybe I should start it with an underscore or something?  The class only exists for DRY.